spdx/parsers/tagvaluebuilders.py

```diff
@@ -892,18 +895,18 @@ def set_pkg_cr_text(self, doc, text):
         Raise value error if text is not one of [None, NOASSERT, TEXT].
         """
         self.assert_package_exists()
-        if not self.package_cr_text_set:
-            self.package_cr_text_set = True
-            if validations.validate_pkg_cr_text(text, optional=True):
-                if isinstance(text, str):
-                    doc.packages[-1].cr_text = str_from_text(text)
-                else:
-                    doc.packages[-1].cr_text = text  # None or NoAssert
-            else:
-                raise SPDXValueError("Package::CopyrightText")
-        else:
+        if self.package_cr_text_set:
             raise CardinalityError("Package::CopyrightText")
 
+        self.package_cr_text_set = True
```

Comment:
In all other cases, you set the "x_set" boolean _after_ validation. Any reason for doing it before validation here?

Reply:
nope, just an oversight